### PR TITLE
Exercise for chapter 17: Classes and Methods

### PR DESCRIPTION
**Pull request by hothanhtuan1209**

### CODE REVIEW

File: exercise/change_attributes.py [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+Modify the methods (and the function int_to_time) to work with the
+new implementation.
+"""
+
+class Time(object):
+    """
+    Represents the time of day.
+    """
+
+    def __init__(self, hour=0, minute=0, second=0):
```
Comment (cuongdoan-agilityio) [inline exercise/change_attributes.py:15]: Update the name for parameters

File: exercise/change_attributes.py [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+        """
+        Initializes a Time object.
+
+        Attributes: hour, minute, second
+        """
+        
+        minutes = hour * 60 + minute
+        self.int_time = (minutes * 60) + second
```
Comment (cuongdoan-agilityio) [inline exercise/change_attributes.py]: Can you get another name for this variable (int_time)?

File: exercise/change_attributes.py [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+        """
+        Returns a string representation of the Time object in the form 'hour:minute:second'.
+
+        Returns:
+            str: A formatted string representing the time.
+        """
+        
+        minutes, second = divmod(self.int_time, 60)
+        hour, minute = divmod(minutes, 60)
+        return "%.2d:%.2d:%.2d" % (hour, minute, second)
```
Comment (cuongdoan-agilityio) [inline exercise/change_attributes.py:35]: Should add the empty line before the return statement.
Double-check for all files in this PR

File: exercise/change_attributes.py [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+
+    def print_time(self):
+        """
+        Prints the time in the form of an integer representing the total seconds.
+        """
+        
+        print(str(self.int_time))
+
+    def time_to_int(self):
```
Comment (cuongdoan-agilityio) [inline exercise/change_attributes.py]: I see you have defined this method but haven't used it yet.

File: exercise/change_attributes.py [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+        assert self.is_valid() and other.is_valid()
+        seconds = self.int_time + other.int_time
+        return seconds
+
+    def increment(self, seconds):
+        """
+        Returns a new Time that is the sum of this time and seconds.
+        """
+
+        return Time(seconds + self.int_time)
```
Comment (cuongdoan-agilityio) [inline exercise/change_attributes.py]: Invalid code.

File: exercise/change_attributes.py [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+class Time(object):
+    """
+    Represents the time of day.
+    """
+
+    def __init__(self, hour=0, minute=0, second=0):
+        """
+        Initializes a Time object.
+
+        Attributes: hour, minute, second
```
Comment (cuongdoan-agilityio) [inline exercise/change_attributes.py]: Should be
```
Attributes:  
    hour: int  
    minute: int  
    second: int  
```
Double-check for all

File: exercise/change_attributes.py [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+        Computes the number of seconds since midnight
+        """
+
+        minutes = self.hour * 60 + self.minute
+        seconds = minutes * 60 + self.second
+        return seconds
+
+    def is_after(self, other):
+        """
+        Returns True if t1 is after t2; false otherwise
```
Comment (cuongdoan-agilityio) [inline exercise/change_attributes.py]: Invalid docstring.

File: exercise/change_attributes.py [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+        """
+
+        if isinstance(other, Time):
+            return self.add_time(other)
+        else:
+            return self.increment(other)
+
+    def __radd__(self, other):
+        """
+        Adds two Time objects or a Time object and a number
```
Comment (cuongdoan-agilityio) [inline exercise/change_attributes.py:74]: Missing information for docstring.

File: exercise/change_attributes.py [hunk 9]
```diff
@@ -0,0 +1,147 @@
...
+        return seconds
+
+    def increment(self, seconds):
+        """
+        Returns a new Time that is the sum of this time and seconds.
+        """
+
+        return Time(seconds + self.int_time)
+
+    def is_valid(self):
```
Comment (cuongdoan-agilityio) [inline exercise/change_attributes.py:95]: Can you check if I entered incorrect data?
Example: Time(-1, 59, 60)

File: exercise/kangaroo_add.py [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+        """
+        Initializes a new Kangaroo object with an empty pouch.
+        """
+        
+        self.pouch_contents = []
+    
+    def put_in_pouch(self, item):
+        """
+        Adds the given item to the pouch_contents list.
+        Args:
```
Comment (cuongdoan-agilityio) [inline exercise/kangaroo_add.py]: What is this mean?

File: exercise/change_attributes.py [hunk 11]
```diff
@@ -0,0 +1,147 @@
...
+    print(start, end)
+
+    start = Time(9, 45)
+    duration = Time(1, 35)
+    print(start + duration)
+    print(start + 1337)
+    print(1337 + start)
+
+    print("Example of polymorphism")
+    t1 = Time(7, 43)
```
Comment (cuongdoan-agilityio) [inline exercise/change_attributes.py]: Update the variable name.

File: exercise/change_attributes.py [hunk 12]
```diff
@@ -0,0 +1,166 @@
...
+            hour (int): The hour of the time (default is 0).
+            minute (int): The minute of the time (default is 0).
+            second (int): The second of the time (default is 0).
+        """
+        
+        self.hour = hour
+        self.minute = minute
+        self.second = second
+        minutes = hour * 60 + minute
+        self.total_seconds = (minutes * 60) + second
```
Comment (cuongdoan-agilityio) [inline exercise/change_attributes.py]: No need ()

File: exercise/change_attributes.py [hunk 13]
```diff
@@ -0,0 +1,166 @@
...
+    print(start + duration)
+    print(start + 1337)
+    print(1337 + start)
+
+    print("Example of polymorphism")
+    time_1 = Time(7, 43)
+    time_2 = Time(7, 41)
+    time_3 = Time(7, 37)
+    total = sum([time_1, time_2, time_3])
+    print(total)
```
Comment (cuongdoan-agilityio) [inline exercise/change_attributes.py]: Please double-check, Calculation results are not as expected.

File: exercise/change_attributes.py (hunk 1 above)
Comment (hothanhtuan1209) [inline exercise/change_attributes.py]: Done

File: exercise/change_attributes.py (hunk 2 above)
Comment (hothanhtuan1209) [reply exercise/change_attributes.py]: Done and remove ()

File: exercise/change_attributes.py (hunk 4 above)
Comment (hothanhtuan1209) [reply exercise/change_attributes.py]: Remove this function

File: exercise/change_attributes.py (hunk 5 above)
Comment (hothanhtuan1209) [reply exercise/change_attributes.py]: Done

File: exercise/change_attributes.py (hunk 6 above)
Comment (hothanhtuan1209) [reply exercise/change_attributes.py]: Done

File: exercise/change_attributes.py (hunk 7 above)
Comment (hothanhtuan1209) [reply exercise/change_attributes.py]: Done

File: exercise/kangaroo_add.py (hunk 10 above)
Comment (hothanhtuan1209) [reply exercise/kangaroo_add.py]: Done

File: exercise/change_attributes.py (hunk 12 above)
Comment (hothanhtuan1209) [reply exercise/change_attributes.py]: Done

File: exercise/change_attributes.py (hunk 11 above)
Comment (hothanhtuan1209) [reply exercise/change_attributes.py]: Done

File: exercise/change_attributes.py (hunk 13 above)
Comment (hothanhtuan1209) [reply exercise/change_attributes.py]: Done

**Pull request merged**